libyul/optimiser/SSAReverser.cpp

```diff
@@ -0,0 +1,55 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#include <libyul/optimiser/SSAReverser.h>
+#include <libyul/AsmData.h>
+#include <libdevcore/CommonData.h>
+
+using namespace std;
+using namespace dev;
+using namespace yul;
+
+void SSAReverser::operator()(Block& _block)
+{
+	walkVector(_block.statements);
+	iterateReplacingMulti<2>(
```

Comment:
We could deduce the ``2`` argument here... but that would require quite some template-magic code, so I'm not sure it's worth it:
```
namespace detail
{

template <typename F, typename... Args>
struct is_invocable: std::is_constructible<
	std::function<void(Args ...)>,
	std::reference_wrapper<typename std::remove_reference<F>::type>
> {};

template<typename F, typename T, std::size_t N, typename... Args>
struct IsCallableWithNtimesT: IsCallableWithNtimesT<F, T, N-1, Args..., T> {};

template<typename F, typename T, typename... Args>
struct IsCallableWithNtimesT<F, T, 0, Args...>: is_invocable<F, Args...> {};

template<typename F, typename T, std::size_t N = 0, typename = void>
struct NumberOfRequiredTs;

template<typename F, typename T, std::size_t N>
struct NumberOfRequiredTs<F, T, N, std::enable_if_t<IsCallableWithNtimesT<F, T, N>::value>>: std::integral_constant<std::size_t, N> {};

template<typename F, typename T, std::size_t N>
struct NumberOfRequiredTs<F, T, N, std::enable_if_t<!IsCallableWithNtimesT<F, T, N>::value && N < 10>>: NumberOfRequiredTs<F, T, N+1> {};

template<typename F, typename T, std::size_t N>
struct NumberOfRequiredTs<F, T, N, std::enable_if_t<!IsCallableWithNtimesT<F, T, N>::value && N >= 10>> {};

}
```

Reply:
I don't think it's worth it. Plus: It saves you 3 characters to type but might cost you a full day debugging where the hell something broke.